app/services/support_interface/notes_export.rb

```diff
@@ -0,0 +1,30 @@
+module SupportInterface
+  class NotesExport
+    def data_for_export
+      notes = Note.select(:subject, :message, :created_at, :application_choice_id, :provider_user_id)
+                .includes(
+                  { provider_user: { provider_permissions: :provider } },
+                  { application_choice: [{ application_form: :candidate }, { course_option: { course: %i[provider accredited_provider] } }] },
+                )
+
+      notes.map do |note|
+        providers = note.provider_user.providers
+        training_provider = note.application_choice.course_option.course.provider
+        ratifying_provider = note.application_choice.course_option.course.accredited_provider
+        provider = providers.include?(training_provider) ? training_provider : ratifying_provider
```

Comment:
Is there a case we have missed here? What if the offered_course was with a different training provider and the user making the note is from the new training provider?
More generally, it might be nice to guarantee we return something from `provider_user.providers` by choosing the first match between that and the list of providers that comes from `course` and `offered_course`

Reply:
I've amended here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4371/commits/6ecfe77267cc0d1528e0b5bea44a03c23eaba27f and added a more complex test scenario where the ratifying provider user is associated to an offered course option.
I don't think we need to sanitise whether the various permeations of providers belong to the note-making provider user as this should always be the case given the nature of the default 'view applications' permission.

Reply:
I was thinking that sometimes providers lose the view applications permission when an application is offered a course that is not run or ratified by their organisation

Reply:
Yeah, it may get confusing which provider is doing what. I think this situation is quite possible and we should at least avoid returning incorrect data here. e.g.

```
training_provider1 = create(:provider)
training_provider2 = create(:provider)
ratifying_provider = create(:provider)

course = create(:course, provider: training_provider1, accredited_provider: ratifying_provider)
offered_course = create(:course, provider: training_provider2, accredited_provider: ratifying_provider)

application_choice = create(:application_choice, candidate: candidate, course_option: create(:course_option, course: course), offered_course_option: create(:course_option, course: offered_course))

provider_user1 = create(:provider_user, providers: [training_provider1])
provider_user2 = create(:provider_user, providers: [training_provider2])

create(:provider_relationship_permissions, training_provider: training_provider1, ratifying_provider: ratifying_provider)
create(:provider_relationship_permissions, training_provider: training_provider2, ratifying_provider: ratifying_provider)

note1 = create(:note, application_choice: application_choice, provider_user: provider_user1)
note2 = create(:note, application_choice: application_choice, provider_user: provider_user2)
```

And then I'd expect the export to do something like:
```
        {
          note_subject: note1.subject,
          note_message: note1.message,
          note_created_at: note1.created_at,
          candidate_id: candidate.id,
          provider_code: training_provider1.code,
          provider_name: training_provider1.name,
          provider_user_id: provider_user1.id,
          number_of_training_provider_relationships: ...,
          total_number_of_provider_relationships: ...,
        },
        {
          note_subject: note2.subject,
          note_message: note2.message,
          note_created_at: note2.created_at,
          candidate_id: candidate.id,
          provider_code: training_provider2.code,
          provider_name: training_provider2.name,
          provider_user_id: provider_user2.id,
          number_of_training_provider_relationships: ...,
          total_number_of_provider_relationships: ...,
        },
```
at the moment the first row will have the ratifying provider details, even though none of the users making notes are from the ratifying org

Reply:
Thanks @benswannack for the test scenario, I've amended the export and included the test case here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4371/commits/066d30bbe8833a488ea5bbd923da7df18570c746 so now we return no provider or org permissions count data in the event of the application becoming disassociated with the note-making user.